extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -235,7 +289,16 @@ export class AmpUserNotification extends AMP.BaseElement {
     this.element.classList.remove('amp-active');
     this.element.classList.add('amp-hidden');
     this.dialogResolve_();
-    this.postDismissEnpoint_();
+
+    // Store and post.
+    if (this.storageKey_) {
+      this.storagePromise_.then(storage => {
```

Comment:
Shouldn't this be done as a `then` callback inside the `storageKey_` conditional in `#shouldShow`?